xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -917,9 +971,34 @@ bool CDVDPlayer::ReadPacket(DemuxPacket*& packet, CDemuxStream*& stream)
       m_OmxPlayerState.bOmxSentEOFs = false;
     }
   }
-  // read a data frame from stream.
-  if(m_pDemuxer)
-    packet = m_pDemuxer->Read();
+
+  int source = m_CurrentVideo.source;
+
+  if (m_CurrentAudio.source > 0 && (m_CurrentAudio.source != m_CurrentVideo.source))
+  {
+    if (m_CurrentVideo.dts > m_CurrentAudio.dts)
```

Comment:
I've done it that way on elupus (cryptic ;)) advice. My first approach used the fill level of the audio player. I'm not very fit in that area, do you have an idea?